src/pages/registration/resource_type_tab/components/JournalField.tsx

```diff
@@ -147,6 +152,14 @@ export const JournalField = ({ confirmedContextType, unconfirmedContextType }: J
             renderOption={(props, option, state) => (
               <PublicationChannelOption key={option.id} props={props} option={option} state={state} />
             )}
+            ListboxComponent={AutocompleteListboxWithExpansion}
+            ListboxProps={
+              {
+                hasMoreHits: !!journalOptionsQuery.data?.totalHits && journalOptionsQuery.data.totalHits > searchSize,
```

Comment:
Kunne denne vært trekt ut i en egen extension metode på useQuery: `someQuery.hasMoreHits = someQuery.totalHits > someQuery.hits.length`? `searchSize` er jo indirekte reflektert i antall treff i resultatet av queryen.

Jeg foreslår å legge til et felt på useQuery som om det er lett gjort, men hvis vinninga går opp i spinning så er det bare å ignorere dette.

Reply:
Har du vært borti å lage extension methods på sånne pakker før?
Jeg har ikke, og kan ikke se de har noe om det i docs heller(?). En mulighet er å lage vår egen wrapper da, som returnerer `useQuery` og støtter noen ekstra ting som vi legger inn, men ser for meg det blir knot å få riktig typer da, ref [denne](https://github.com/BIBSYSDEV/NVA-Frontend/blob/c8cf785d4c82802ef1f0b802a6504fadd79aee6e/src/components/AutocompleteListboxWithExpansion.tsx#L6) feks, hvor jeg ikke fikk til å bare ta en Pick av props fra typen til `useQuery` heller fordi den avhenger av så mye rart 🤔  Mulig at jeg har misset noe her altså

Reply:
Ser ikke ut som TypeScript har noe begrep om dette, og at det nærmeste man kommer er hjelpemetoder. Det kan gjøre det litt mer leselig, men er ikke like clean som jeg hadde håpet :/